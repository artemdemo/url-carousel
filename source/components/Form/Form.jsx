import React from 'react';

const Form = (props) => {
    const { className } = props;
    return (
        <form
            className={className}
            onSubmit={(e) => {
                const { onSubmit } = props;
                e.preventDefault();
                onSubmit && onSubmit();
            }}>
            {props.children}
        </form>
    );
};

Form.propTypes = {
    onSubmit: React.PropTypes.func,
    className: React.PropTypes.string,
};

export default Form;
