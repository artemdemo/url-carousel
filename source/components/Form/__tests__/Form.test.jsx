import React from 'react';
import renderer from 'react-test-renderer';
import toJSON from 'enzyme-to-json';
import { mount } from 'enzyme';
import Form from '../Form';

describe('<Form>', () => {
    it('Render', () => {
        const tree = renderer.create(
            <Form>
                <input />
            </Form>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('with className', () => {
        const tree = renderer.create(
            <Form className='form-class'>
                <input />
            </Form>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handle submit', () => {
        let submitted = false;
        const wrapper = mount(
            <Form onSubmit={() => submitted = true}>
                <input />
            </Form>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.simulate('submit');
        expect(submitted).toEqual(true);
    });
});
