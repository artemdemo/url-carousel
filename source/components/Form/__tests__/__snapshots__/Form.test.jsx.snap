// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> Render 1`] = `
<form
  className={undefined}
  onSubmit={[Function]}
>
  <input />
</form>
`;

exports[`<Form> handle submit 1`] = `
<Form
  onSubmit={[Function]}
>
  <form
    onSubmit={[Function]}
  >
    <input />
  </form>
</Form>
`;

exports[`<Form> with className 1`] = `
<form
  className="form-class"
  onSubmit={[Function]}
>
  <input />
</form>
`;
